tests/python/test_python_frontend.py

```diff
@@ -2402,6 +2402,51 @@ def fusion_func(fd: FusionDefinition, inps) -> None:
             fp16_nvf_out = nvf_out[0]
             self.assertEqual(eager_out, fp16_nvf_out)
 
+    def test_linear(self):
+        m = 24
+        n = 16
+        k = 8
+        bias = torch.randn(n, device="cuda", dtype=torch.float16)
+        inputs_mk_nk = [
+            torch.randn(m, k, device="cuda", dtype=torch.float16),
+            torch.randn(n, k, device="cuda", dtype=torch.float16),
+        ]
+
+        inputs_mk_kn = [
+            inputs_mk_nk[0].clone(),
+            inputs_mk_nk[1].clone().as_strided(size=[n, k], stride=[1, n]),
+        ]
+
+        inputs_km_nk = [
+            inputs_mk_nk[0].clone().as_strided(size=[m, k], stride=[1, m]),
+            inputs_mk_nk[1].clone(),
+        ]
+
+        inputs_km_kn = [
+            inputs_km_nk[0].clone(),
+            inputs_mk_kn[1].clone(),
+        ]
+
+        def fusion_func(fd: FusionDefinition, inp: torch.Tensor, wt: torch.Tensor, bias: torch.Tensor | None) -> None:
+            t0 = fd.from_pytorch(inp)
+            t1 = fd.from_pytorch(wt)
+            if bias is not None:
+                t2 = fd.from_pytorch(bias)
+                t_out = fd.ops.linear(t0, t1, t2)
+            else:
+                t_out = fd.ops.linear(t0, t1)
+            fd.add_output(t_out)
+
+        for inp, wt in [inputs_mk_nk, inputs_mk_kn, inputs_km_nk, inputs_km_kn]:
+            for use_bias in [True, False]:
```

Comment:
Instead of for loops, I'd parameterize the test: https://docs.pytest.org/en/7.1.x/how-to/parametrize.html. This way, when an error shows up, it's clear from the test name which config is failing.

Reply:
I'm no longer sure we run these tests via pytest. If we don't, unittest has a way to create subtests, replacing iterations: https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests

Reply:
Thank @wujingyue, please take another look at the test.